lib/debug/server_dap.rb

```diff
@@ -1010,16 +1010,6 @@ def evaluate_result r
       variable nil, r
     end
 
-    def type_name obj
-      klass = M_CLASS.bind_call(obj)
-
-      begin
-        klass.name || klass.to_s
-      rescue Exception => e
```

Comment:
why not drop `'klass.to_s`?

Reply:
No. `klass.name` can return `nil`.

Reply:
Ohh, I see. Thanks for the catch! 
I updated the code to use `to_s` and added a test for anonymous class instances to ensure the `type` is `""`.

Reply:
Do you really need `nil` for the name?
Previous one is `klass.name || klass.to_s` but your code uses `klass.name.to_s`.

Reply:
The previous code means that if `klass.name` is given, use it, otherwise use `klass.to_s`.

Reply:
In this case, the error because of `to_s` is ignored.
I'll fix it after merging.